In [134]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns
import numpy as np
import json
import math
import os
from sklearn.preprocessing import LabelEncoder
from sklearn.linear_model import LinearRegression, Ridge
from omegaconf import OmegaConf

In [135]:
pd.set_option('display.max_columns', None)  #Setting the output view to show all the columns

# Data Preprocessing
The data from the .pkl files is imported and read as df0, df1,..,df6 for seven datasets. .head(-2) is used to see the top five and bottom five rows of the dataset to understand the consistentency in the data.

## Configuration of Directory Paths

In [136]:
# Read pickles from the path specified in config.dest_file
config_path = "config.yaml"  # Replace with the path to your config file
config = OmegaConf.load(config_path)

## df0

In [137]:
# Construct the full path to the pickle file
pickle_path = os.path.join(config.dest_file, "dataset.pkl")
df0 = pd.read_pickle(pickle_path)
df0.head(-5)

surfaceProcessingLocationCavity hotRunner hotRunnerCost  \
0      55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         40000   
1      55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         40000   
2      55c4af00-0439-40aa-ab0a-4f1a0fa16882         0         40000   
3      55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         10000   
4      55c4af00-0439-40aa-ab0a-4f1a0fa16882         0         25000   
...                                     ...       ...           ...   
11654  55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         25000   
11655  55c4af00-0439-40aa-ab0a-4f1a0fa16882         0         25000   
11656  55c4af00-0439-40aa-ab0a-4f1a0fa16882         0         25000   
11657  55c4af00-0439-40aa-ab0a-4f1a0fa16882         0         25000   
11658  55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         10000   

      manufacturingCost mouldValidationCost designTime hourlyDesignCost  \
0                 40000                1500         40               65   
1                 25000                3000         70               50   
2                 40000                4500        100               65   
3                 25000                4500         70               50   
4                 25000                3000         40               65   
...                 ...                 ...        ...              ...   
11654             25000                4500         70               65   
11655             25000                3000         70               50   
11656             25000                4500         40               50   
11657             40000                4500         40               50   
11658             55000                4500        100               50   

      removedChassis weightChassisProcessed  \
0               0.20                 6.6725   
1               0.15                 6.6725   
2               0.20                 6.6725   
3               0.18                 6.6725   
4               0.18                 6.6725   
...              ...                    ...   
11654           0.20                 6.6725   
11655           0.15                 6.6725   
11656           0.18                 6.6725   
11657           0.18                 6.6725   
11658           0.15                 6.6725   

           surfaceProcessingLocationChassis removedCavity numberOfCavities  \
0      35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.5               32   
1      35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.5               16   
2      35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.3               32   
3      35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.5               32   
4      35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.3               32   
...                                     ...           ...              ...   
11654  35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.5               32   
11655  35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.5                4   
11656  35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.5                4   
11657  35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.3                4   
11658  35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.5                4   

      weightCavityProcessed weightMould totalTransportationPercentage  \
0      0.005495000000000001    6.677995                           100   
1      0.005495000000000001    6.677995                           100   
2      0.005495000000000001    6.677995                           100   
3      0.005495000000000001    6.677995                           100   
4      0.005495000000000001    6.677995                           100   
...                     ...         ...                           ...   
11654  0.005495000000000001    6.677995                           100   
11655  0.005495000000000001    6.677995                           100   
11656  0.005495000000000001    6.677995                           100   
11657  0.00549500000000000

### Explanation 

The data has consistent data enteries and hence can be used.

## df1

In [138]:
# Construct the full path to the pickle file
pickle_path = os.path.join(config.dest_file, "dataset_1.pkl")
df1 = pd.read_pickle(pickle_path)
df1.head(-5)

nProdottiAnno Cost designTime  \
0      55c4af00-0439-40aa-ab0a-4f1a0fa16882    0      10000   
1      55c4af00-0439-40aa-ab0a-4f1a0fa16882    0      10000   
2      55c4af00-0439-40aa-ab0a-4f1a0fa16882    0      10000   
3      55c4af00-0439-40aa-ab0a-4f1a0fa16882    0      10000   
4      55c4af00-0439-40aa-ab0a-4f1a0fa16882    0      10000   
...                                     ...  ...        ...   
30608  55c4af00-0439-40aa-ab0a-4f1a0fa16882    0      10000   
30609  55c4af00-0439-40aa-ab0a-4f1a0fa16882    0      10000   
30610  55c4af00-0439-40aa-ab0a-4f1a0fa16882    0      10000   
30611  55c4af00-0439-40aa-ab0a-4f1a0fa16882    0      10000   
30612  55c4af00-0439-40aa-ab0a-4f1a0fa16882    0      10000   

      ecosystem quality - land occupation  \
0                                   55000   
1                                   40000   
2                                   55000   
3                                   25000   
4                                   55000   
...                                   ...   
30608                               55000   
30609                               25000   
30610                               25000   
30611                               25000   
30612                               40000   

      human health - ozone layer depletion human health - ionising radiation  \
0                                     1500                                40   
1                                     1500                                40   
2                                     1500                                40   
3                                     1500                                40   
4                                     1500                                40   
...                                    ...                               ...   
30608                                 1500                                40   
30609                                 1500                                40   
30610                                 1500                                40   
30611                                 1500                                40   
30612                                 1500                                40   

      injectedMaterialCost maintenanceCost human health - total  \
0                       50            0.15               6.6725   
1                       50            0.15               6.6725   
2                       50            0.15               6.6725   
3                       50            0.15               6.6725   
4                       50            0.15               6.6725   
...                    ...             ...                  ...   
30608                   50            0.15               6.6725   
30609                   50            0.15               6.6725   
30610                   50            0.15               6.6725   
30611                   50            0.15               6.6725   
30612                   50            0.15               6.6725   

                           numberOfCavities warmupTime cycleTime  \
0      35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3         1   
1      35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3         1   
2      35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3         1   
3      35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3         1   
4      35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3         1   
...                                     ...        ...       ...   
30608  35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3         1   
30609  35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3         1   
30610  35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3         1   
30611  35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3         1   
30612  35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3         1   

          mouldMaterialName climate change - total transportCost  \
0      0.005495000000000001               6.677995           100   
1      0.0054

## df2

In [139]:
pickle_path = os.path.join(config.dest_file, "dataset_2.pkl")
df2 = pd.read_pickle(pickle_path)
df2.head(-8501)

resources - non-renewable energy removedChassis machineCycleTime  \
0      55c4af00-0439-40aa-ab0a-4f1a0fa16882              0            10000   
1      55c4af00-0439-40aa-ab0a-4f1a0fa16882              0            10000   
2      55c4af00-0439-40aa-ab0a-4f1a0fa16882              0            10000   
3      55c4af00-0439-40aa-ab0a-4f1a0fa16882              0            10000   
4      55c4af00-0439-40aa-ab0a-4f1a0fa16882              0            10000   
...                                     ...            ...              ...   
11663  55c4af00-0439-40aa-ab0a-4f1a0fa16882              0            10000   
11664                              0.001789              1               10   
11665                              0.002491              1               10   
11666                               0.00223              1               10   
11667                               0.00185              1               10   

      deliveryVolume maintenanceCost resources - mineral extraction  \
0              25000            1500                             40   
1              25000            1500                             40   
2              40000            1500                             40   
3              40000            1500                             40   
4              55000            1500                             40   
...              ...             ...                            ...   
11663          55000            1500                             40   
11664            100              25                       0.000018   
11665            100              25                       0.000019   
11666            100              25                       0.000019   
11667            100              25                       0.000018   

      ecosystem quality - terrestrial acidification & nutrification  \
0                                                     50              
1                                                     50              
2                                                     50              
3                                                     50              
4                                                     50              
...                                                  ...              
11663                                                 50              
11664                                                  0              
11665                                                  0              
11666                                                  0              
11667                                                  0              

      steelPrice human health - ozone layer depletion  \
0           0.15                               6.6725   
1           0.15                               6.6725   
2           0.15                               6.6725   
3           0.15                               6.6725   
4           0.15                               6.6725   
...          ...                                  ...   
11663       0.15                               6.6725   
11664          0                         69493.360588   
11665          0                         54493.360588   
11666          0                         82493.360588   
11667          0                         65493.360588   

                             deliveryPeriod gateDiameter mouldValidationCost  \
0      35e16aba-caff-4fc6-a8ed-ec620c0da77f          0.3                  32   
1      35e16aba-caff-4fc6-a8ed-ec620c0da77f          0.3                  32   
2      35e16aba-caff-4fc6-a8ed-ec620c0da77f          0.3                  32   
3      35e16aba-caff-4fc6-a8ed-ec620c0da77f          0.3                  16   
4      35e16aba-caff-4fc6-a8ed-ec620c0da77f          0.3                   4   
...                                     ...          ...                 ...   
11663  35e16aba-caff-4fc6-a8ed-ec620c0da77f          0.3                  32   
11664  8f10d917-79d3-48e6-a834-

The above df2 has inconsistent data enteries. Upon further investigating, It has been found that the data has been shifted a few columns. It is important to identify at which column it happened. It has been found to be at row index 11664. hence rows after index 11663 have been dropped since they will not effect the overall size of our df.


In [140]:
df2_selected = df2.iloc[:11666]

In [141]:
df2_selected.head(-2)

resources - non-renewable energy removedChassis machineCycleTime  \
0      55c4af00-0439-40aa-ab0a-4f1a0fa16882              0            10000   
1      55c4af00-0439-40aa-ab0a-4f1a0fa16882              0            10000   
2      55c4af00-0439-40aa-ab0a-4f1a0fa16882              0            10000   
3      55c4af00-0439-40aa-ab0a-4f1a0fa16882              0            10000   
4      55c4af00-0439-40aa-ab0a-4f1a0fa16882              0            10000   
...                                     ...            ...              ...   
11659  55c4af00-0439-40aa-ab0a-4f1a0fa16882              0            10000   
11660  55c4af00-0439-40aa-ab0a-4f1a0fa16882              0            10000   
11661  55c4af00-0439-40aa-ab0a-4f1a0fa16882              0            10000   
11662  55c4af00-0439-40aa-ab0a-4f1a0fa16882              0            10000   
11663  55c4af00-0439-40aa-ab0a-4f1a0fa16882              0            10000   

      deliveryVolume maintenanceCost resources - mineral extraction  \
0              25000            1500                             40   
1              25000            1500                             40   
2              40000            1500                             40   
3              40000            1500                             40   
4              55000            1500                             40   
...              ...             ...                            ...   
11659          40000            1500                             40   
11660          55000            1500                             40   
11661          55000            1500                             40   
11662          40000            1500                             40   
11663          55000            1500                             40   

      ecosystem quality - terrestrial acidification & nutrification  \
0                                                     50              
1                                                     50              
2                                                     50              
3                                                     50              
4                                                     50              
...                                                  ...              
11659                                                 50              
11660                                                 50              
11661                                                 50              
11662                                                 50              
11663                                                 50              

      steelPrice human health - ozone layer depletion  \
0           0.15                               6.6725   
1           0.15                               6.6725   
2           0.15                               6.6725   
3           0.15                               6.6725   
4           0.15                               6.6725   
...          ...                                  ...   
11659       0.15                               6.6725   
11660       0.15                               6.6725   
11661       0.15                               6.6725   
11662       0.15                               6.6725   
11663       0.15                               6.6725   

                             deliveryPeriod gateDiameter mouldValidationCost  \
0      35e16aba-caff-4fc6-a8ed-ec620c0da77f          0.3                  32   
1      35e16aba-caff-4fc6-a8ed-ec620c0da77f          0.3                  32   
2      35e16aba-caff-4fc6-a8ed-ec620c0da77f          0.3                  32   
3      35e16aba-caff-4fc6-a8ed-ec620c0da77f          0.3                  16   
4      35e16aba-caff-4fc6-a8ed-ec620c0da77f          0.3                   4   
...                                     ...          ...                 ...   
11659  35e16aba-caff-4fc6-a8ed-ec620c0da77f          0.3                  16   
11660  35e16aba-caff-4fc6-a8ed-

We have saved the clipped dataframe df2


In [142]:
df2=df2_selected

## df3


In [143]:
pickle_path = os.path.join(config.dest_file, "dataset_3.pkl")
df3= pd.read_pickle(pickle_path)
df3.head(2)

maintenanceCost totalTransportationPercentage  \
0  55c4af00-0439-40aa-ab0a-4f1a0fa16882                             0   
1  55c4af00-0439-40aa-ab0a-4f1a0fa16882                             0   

  percentageLorry deliveryVolume removedCavity cycleTime human health - total  \
0           10000          55000          1500        40                   50   
1           10000          55000          1500        40                   50   

  climate change - total EUUSMacchina              injectionMouldingProcess  \
0                   0.15       6.6725  35e16aba-caff-4fc6-a8ed-ec620c0da77f   
1                   0.15       6.6725  35e16aba-caff-4fc6-a8ed-ec620c0da77f   

  materozzaVolume CNMacchina                  Cost  \
0             0.3          1  0.005495000000000001   
1             0.3          1  0.005495000000000001   

  resources - mineral extraction ecosystem quality - total numberOfCavities  \
0                       6.677995                       100                5   
1                       6.677995                       100                5   

  machineCycleTime hotRunnerCost mouldMaterialName mouldTotalCost  \
0               25            40                30           6850   
1               25            40                30           3450   

  resources - non-renewable energy                     resources - total  \
0                             4000  be308d91-fab4-43c9-bc66-726c5b033b86   
1                             4000  932e3143-0f29-4b07-a98a-8cd9507ec5d8   

  designTime ecosystem quality - aquatic ecotoxicity  \
0          9                                       1   
1          5                                       1   

  surfaceProcessingLocationCavity productVolume percentageRecycledMaterial  \
0                              50             3                         35   
1                               5             3                         35   

  totalDistance surfaceProcessingLocationChassis weightChassisProcessed  \
0             1                                1                      1   
1             1                                1                      1   

  removedChassis gateDiameter weightCavityProcessed weightMould runnersType  \
0        0.00095            0                     0         360          10   
1        0.00123            0                     0         360          10   

  injectedMaterial_product maxDepth                mouldDesignCostDisplay  \
0                     8000       12  be308d91-fab4-43c9-bc66-726c5b033b86   
1                     8000       12  932e3143-0f29-4b07-a98a-8cd9507ec5d8   

                              setupTime    percentageShip  \
0  5984ad89-00c1-446d-880c-dad9bbb81f02  094e332c4dcaacfd   
1  5984ad89-00c1-446d-880c-dad9bbb81f02  094e332c4dcaacfd   

  ecosystem quality - terrestrial acidification & nutrification  \
0                                                 -1              
1                                                 -1              

  surfaceFinishing ecosystem quality - terrestrial ecotoxicity  \
0             COLD                            95% PP + 5% PTFE   
1             COLD                          90% PA6 + 10% PTFE   

           hotRunner    percentageTrain injectedMaterialCost maxWallThickness  \
0  Horizontal 10 ton  Horizontal 10 ton    Horizontal 10 ton                1   
1  Horizontal 10 ton  Horizontal 10 ton    Horizontal 10 ton                1   

  human health - ionising radiation tolerance materialDensity steelPrice  \
0                                15        30              10          2   
1                                15        30              10          2   

  climate change - climate change ecosystem quality - land occupation  \
0                            2000                               38500   
1                            2000                               38500   

   nAnniProduzione  memtiEngineValue  mouldValidationCost  nProdottiAnno  \
0     80493.654588          0.

## df4


In [144]:
pickle_path = os.path.join(config.dest_file, "dataset_4.pkl")
df4= pd.read_pickle(pickle_path)
df4.head(-2)

weightMould  \
0      55c4af00-0439-40aa-ab0a-4f1a0fa16882   
1      55c4af00-0439-40aa-ab0a-4f1a0fa16882   
2      55c4af00-0439-40aa-ab0a-4f1a0fa16882   
3      55c4af00-0439-40aa-ab0a-4f1a0fa16882   
4      55c4af00-0439-40aa-ab0a-4f1a0fa16882   
...                                     ...   
12621  55c4af00-0439-40aa-ab0a-4f1a0fa16882   
12622  55c4af00-0439-40aa-ab0a-4f1a0fa16882   
12623  55c4af00-0439-40aa-ab0a-4f1a0fa16882   
12624  55c4af00-0439-40aa-ab0a-4f1a0fa16882   
12625  55c4af00-0439-40aa-ab0a-4f1a0fa16882   

      ecosystem quality - terrestrial ecotoxicity maintenanceCost hotRunner  \
0                                               0           10000     25000   
1                                               0           10000     25000   
2                                               0           10000     25000   
3                                               0           10000     25000   
4                                               0           10000     25000   
...                                           ...             ...       ...   
12621                                           0           10000     25000   
12622                                           0           10000     25000   
12623                                           0           10000     25000   
12624                                           0           10000     25000   
12625                                           0           10000     25000   

      tolerance removedCavity weightCavityProcessed  \
0          1500            40                    50   
1          1500            40                    50   
2          1500            40                    50   
3          1500            40                    50   
4          1500            40                    50   
...         ...           ...                   ...   
12621      1500            40                    50   
12622      1500            40                    50   
12623      1500            40                    50   
12624      1500            40                    50   
12625      1500            40                    50   

      percentageRecycledMaterial designTime  \
0                           0.15     6.6725   
1                           0.15     6.6725   
2                           0.15     6.6725   
3                           0.15     6.6725   
4                           0.15     6.6725   
...                          ...        ...   
12621                       0.15     6.6725   
12622                       0.15     6.6725   
12623                       0.15     6.6725   
12624                       0.15     6.6725   
12625                       0.15     6.6725   

                                machineName warmupTime  \
0      35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3   
1      35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3   
2      35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3   
3      35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3   
4      35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3   
...                                     ...        ...   
12621  35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3   
12622  35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3   
12623  35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3   
12624  35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3   
12625  35e16aba-caff-4fc6-a8ed-ec620c0da77f        0.3   

      ecosystem quality - aquatic ecotoxicity  \
0                                           1   
1                                           1   
2                                           1   
3                                           1   
4                                           1   
...                                       ...   
12621                                       1   
12622                                       1   
12623                                       1   
12624                                       1   
12625                                       1   

      

## df5

In [145]:
pickle_path = os.path.join(config.dest_file, "dataset_5.pkl")
df5 = pd.read_pickle(pickle_path)
df5.head(-2)

hourlyDesignCost warmupTime mouldTotalCost  \
0      7c09041d-35fc-4e1a-905f-7499e3a81a8f          0          10000   
1      55c4af00-0439-40aa-ab0a-4f1a0fa16882          0          10000   
2      1c1e35f0-6355-42f5-92b4-8081fcc865f2          0          10000   
3      7c09041d-35fc-4e1a-905f-7499e3a81a8f          0          10000   
4      7c09041d-35fc-4e1a-905f-7499e3a81a8f          0          10000   
...                                     ...        ...            ...   
23321  7c09041d-35fc-4e1a-905f-7499e3a81a8f          0          10000   
23322  55c4af00-0439-40aa-ab0a-4f1a0fa16882          0          10000   
23323  55c4af00-0439-40aa-ab0a-4f1a0fa16882          0          10000   
23324  7c09041d-35fc-4e1a-905f-7499e3a81a8f          0          10000   
23325  7c09041d-35fc-4e1a-905f-7499e3a81a8f          0          10000   

      climate change - climate change percentageAircraft cycleTime  \
0                               25000               1500        40   
1                               25000               1500        40   
2                               25000               1500        40   
3                               25000               1500        40   
4                               25000               1500        40   
...                               ...                ...       ...   
23321                           25000               1500        40   
23322                           25000               1500        40   
23323                           25000               1500        40   
23324                           25000               1500        40   
23325                           25000               1500        40   

      mouldMaterialName resources - non-renewable energy machineName  \
0                    50                             0.15      6.6725   
1                    50                             0.15      6.6725   
2                    50                             0.15      6.6725   
3                    50                             0.15      6.6725   
4                    50                             0.15      6.6725   
...                 ...                              ...         ...   
23321                50                             0.15      6.6725   
23322                50                             0.15      6.6725   
23323                50                             0.15      6.6725   
23324                50                             0.15      6.6725   
23325                50                             0.15      6.6725   

                                 CNMacchina resources - mineral extraction  \
0      d65ad7e8-09d9-47b7-a1cb-9e3c5460b9a2                            0.3   
1      35e16aba-caff-4fc6-a8ed-ec620c0da77f                            0.3   
2      e52c3045-7a2a-44f7-9a3e-8ecc23d96345                            0.3   
3      d65ad7e8-09d9-47b7-a1cb-9e3c5460b9a2                            0.3   
4      d65ad7e8-09d9-47b7-a1cb-9e3c5460b9a2                            0.3   
...                                     ...                            ...   
23321  d65ad7e8-09d9-47b7-a1cb-9e3c5460b9a2                            0.3   
23322  35e16aba-caff-4fc6-a8ed-ec620c0da77f                            0.3   
23323  35e16aba-caff-4fc6-a8ed-ec620c0da77f                            0.3   
23324  d65ad7e8-09d9-47b7-a1cb-9e3c5460b9a2                            0.3   
23325  d65ad7e8-09d9-47b7-a1cb-9e3c5460b9a2                            0.3   

      ecosystem quality - land occupation injectedMaterial_product  \
0                                       1     0.005495000000000001   
1                                       1     0.005495000000000001   
2                                       1     0.005495000000000001   
3                                       1     0.005495000000000001   
4                                       1     0.005495000000000001   
...                                   ...                      ...   
23321            

## df6

In [146]:
pickle_path = os.path.join(config.dest_file, "dataset_6.pkl")
df6 = pd.read_pickle(pickle_path)
df6.head(-2)

machineCycleTime human health - human toxicity  \
0      55c4af00-0439-40aa-ab0a-4f1a0fa16882                             0   
1      1c1e35f0-6355-42f5-92b4-8081fcc865f2                             1   
2      55c4af00-0439-40aa-ab0a-4f1a0fa16882                             1   
3      7c09041d-35fc-4e1a-905f-7499e3a81a8f                             0   
4      55c4af00-0439-40aa-ab0a-4f1a0fa16882                             0   
...                                     ...                           ...   
52481  55c4af00-0439-40aa-ab0a-4f1a0fa16882                             1   
52482  7c09041d-35fc-4e1a-905f-7499e3a81a8f                             0   
52483  55c4af00-0439-40aa-ab0a-4f1a0fa16882                             1   
52484  55c4af00-0439-40aa-ab0a-4f1a0fa16882                             0   
52485  1c1e35f0-6355-42f5-92b4-8081fcc865f2                             0   

      mouldDesignCostDisplay climate change - total percentageAircraft  \
0                      25000                  40000               1500   
1                      10000                  25000               1500   
2                      40000                  55000               1500   
3                      25000                  40000               1500   
4                      25000                  25000               1500   
...                      ...                    ...                ...   
52481                  10000                  25000               1500   
52482                  40000                  55000               1500   
52483                  10000                  25000               1500   
52484                  10000                  40000               1500   
52485                  10000                  55000               1500   

      injectedMaterial_product EUUSMacchina designTime hotRunner  \
0                           40           50      0.225    6.6725   
1                           70           50       0.18    6.6725   
2                          100           50       0.18    6.6725   
3                          100           50      0.225    6.6725   
4                           70           50      0.225    6.6725   
...                        ...          ...        ...       ...   
52481                       40           50       0.15    6.6725   
52482                       70           50       0.18    6.6725   
52483                      100           50       0.18    6.6725   
52484                       40           50       0.15    6.6725   
52485                      100           50      0.225    6.6725   

                           numberOfCavities human health - ionising radiation  \
0      35e16aba-caff-4fc6-a8ed-ec620c0da77f                               0.3   
1      e52c3045-7a2a-44f7-9a3e-8ecc23d96345                               0.3   
2      35e16aba-caff-4fc6-a8ed-ec620c0da77f                               0.3   
3      d65ad7e8-09d9-47b7-a1cb-9e3c5460b9a2                               0.3   
4      35e16aba-caff-4fc6-a8ed-ec620c0da77f                               0.3   
...                                     ...                               ...   
52481  35e16aba-caff-4fc6-a8ed-ec620c0da77f                               0.3   
52482  d65ad7e8-09d9-47b7-a1cb-9e3c5460b9a2                               0.3   
52483  35e16aba-caff-4fc6-a8ed-ec620c0da77f                               0.3   
52484  35e16aba-caff-4fc6-a8ed-ec620c0da77f                               0.3   
52485  e52c3045-7a2a-44f7-9a3e-8ecc23d96345                               0.3   

      runnersType resources - non-renewable energy percentageTrain  \
0               1             0.005495000000000001        6.677995   
1               1             0.005495000000000001        6.677995   
2               1             0.005495000000000001        6.677995   
3               1             0.005495000000000001        6.677995   
4               1             0.005495000000000001        6.677995   
...

# Concatenation of dfs in final df

## Consistency of SHAPES all the dfs

In [147]:
# Check the number of rows for each DataFrame
for i, df in enumerate([df0, df1, df2, df3, df4, df5, df6]):
    print(f"Number of rows in df{i}:", df.shape[0], " &  Number of columns:", df.shape[1])


Number of rows in df0: 11664  &  Number of columns: 70
Number of rows in df1: 30618  &  Number of columns: 70
Number of rows in df2: 11666  &  Number of columns: 70
Number of rows in df3: 21758  &  Number of columns: 70
Number of rows in df4: 12628  &  Number of columns: 70
Number of rows in df5: 23328  &  Number of columns: 70
Number of rows in df6: 52488  &  Number of columns: 70


### renaming columns

We also found that the column names are also drifted in df1 onwards, while the data enteries in the respective cells are always the same. So we rename columns of all the dfs based on the first df: df0

In [148]:
# prompt: change df1,df2,.., df6 column names to df0's

import pandas as pd

# Assuming df0, df1, ..., df6 are already defined pandas DataFrames

# Get the column names from df0
new_column_names = df0.columns.tolist()

# Iterate through the dataframes and rename their columns
for i in range(1, 7):
  try:
    globals()[f'df{i}'].columns = new_column_names
  except KeyError:
    print(f"DataFrame df{i} not found.")
  except AttributeError:
    print(f"df{i} is not a DataFrame")

In [149]:
# prompt: concat df0,to df6 without considering coulumn names in df1 to df6

import pandas as pd

# Assuming df0, df1, ..., df6 are already defined DataFrames

# Concatenate the DataFrames, ignoring column names in df1 to df6
concatenated_df = pd.concat([df0] + [df.reset_index(drop=True) for df in [df1, df2, df3, df4, df5, df6]], ignore_index=True)


# Now concatenated_df holds the combined data
concatenated_df



surfaceProcessingLocationCavity hotRunner hotRunnerCost  \
0       55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         40000   
1       55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         40000   
2       55c4af00-0439-40aa-ab0a-4f1a0fa16882         0         40000   
3       55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         10000   
4       55c4af00-0439-40aa-ab0a-4f1a0fa16882         0         25000   
...                                      ...       ...           ...   
164145  55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         10000   
164146  55c4af00-0439-40aa-ab0a-4f1a0fa16882         0         10000   
164147  1c1e35f0-6355-42f5-92b4-8081fcc865f2         0         10000   
164148  1c1e35f0-6355-42f5-92b4-8081fcc865f2         1         40000   
164149  1c1e35f0-6355-42f5-92b4-8081fcc865f2         0         25000   

       manufacturingCost mouldValidationCost designTime hourlyDesignCost  \
0                  40000                1500         40               65   
1                  25000                3000         70               50   
2                  40000                4500        100               65   
3                  25000                4500         70               50   
4                  25000                3000         40               65   
...                  ...                 ...        ...              ...   
164145             25000                1500        100               50   
164146             40000                1500         40               50   
164147             55000                1500        100               50   
164148             55000                1500         70               50   
164149             25000                1500         70               50   

       removedChassis weightChassisProcessed  \
0                0.20                 6.6725   
1                0.15                 6.6725   
2                0.20                 6.6725   
3                0.18                 6.6725   
4                0.18                 6.6725   
...               ...                    ...   
164145           0.18                 6.6725   
164146           0.15                 6.6725   
164147          0.225                 6.6725   
164148           0.15                 6.6725   
164149          0.225                 6.6725   

            surfaceProcessingLocationChassis removedCavity numberOfCavities  \
0       35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.5               32   
1       35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.5               16   
2       35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.3               32   
3       35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.5               32   
4       35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.3               32   
...                                      ...           ...              ...   
164145  35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.3                1   
164146  35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.3                1   
164147  e52c3045-7a2a-44f7-9a3e-8ecc23d96345           0.3                1   
164148  e52c3045-7a2a-44f7-9a3e-8ecc23d96345           0.3                1   
164149  e52c3045-7a2a-44f7-9a3e-8ecc23d96345           0.3                1   

       weightCavityProcessed weightMould totalTransportationPercentage  \
0       0.005495000000000001    6.677995                           100   
1       0.005495000000000001    6.677995                           100   
2       0.005495000000000001    6.677995                           100   
3       0.005495000000000001    6.677995                           100   
4       0.005495000000000001    6.677995                           100   
...                      ...         ...                           ...   
164145  0.005495000000000001    6.677995                           100   
164146  0.005495000000000001    6.677995                           100   
164147  0.005495000000000001    6.677995   

In [150]:
concatenated_df.head(11666)

surfaceProcessingLocationCavity hotRunner hotRunnerCost  \
0      55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         40000   
1      55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         40000   
2      55c4af00-0439-40aa-ab0a-4f1a0fa16882         0         40000   
3      55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         10000   
4      55c4af00-0439-40aa-ab0a-4f1a0fa16882         0         25000   
...                                     ...       ...           ...   
11661  55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         40000   
11662  55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         25000   
11663  55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         25000   
11664  55c4af00-0439-40aa-ab0a-4f1a0fa16882         0         10000   
11665  55c4af00-0439-40aa-ab0a-4f1a0fa16882         0         10000   

      manufacturingCost mouldValidationCost designTime hourlyDesignCost  \
0                 40000                1500         40               65   
1                 25000                3000         70               50   
2                 40000                4500        100               65   
3                 25000                4500         70               50   
4                 25000                3000         40               65   
...                 ...                 ...        ...              ...   
11661             25000                1500         70               65   
11662             40000                4500         70               65   
11663             55000                4500        100               65   
11664             55000                1500         40               50   
11665             40000                1500         40               50   

      removedChassis weightChassisProcessed  \
0               0.20                 6.6725   
1               0.15                 6.6725   
2               0.20                 6.6725   
3               0.18                 6.6725   
4               0.18                 6.6725   
...              ...                    ...   
11661           0.15                 6.6725   
11662           0.18                 6.6725   
11663           0.15                 6.6725   
11664           0.15                 6.6725   
11665           0.15                 6.6725   

           surfaceProcessingLocationChassis removedCavity numberOfCavities  \
0      35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.5               32   
1      35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.5               16   
2      35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.3               32   
3      35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.5               32   
4      35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.3               32   
...                                     ...           ...              ...   
11661  35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.3                4   
11662  35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.3                4   
11663  35e16aba-caff-4fc6-a8ed-ec620c0da77f          0.45                1   
11664  35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.3                1   
11665  35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.3                1   

      weightCavityProcessed weightMould totalTransportationPercentage  \
0      0.005495000000000001    6.677995                           100   
1      0.005495000000000001    6.677995                           100   
2      0.005495000000000001    6.677995                           100   
3      0.005495000000000001    6.677995                           100   
4      0.005495000000000001    6.677995                           100   
...                     ...         ...                           ...   
11661  0.005495000000000001    6.677995                           100   
11662  0.005495000000000001    6.677995                           100   
11663  0.005495000000000001    6.677995                           100   
11664  0.00549500000000000

In [151]:
# Concatenate the DataFrames
df_combined = pd.concat([df0, df1, df2, df3, df4, df5, df6], ignore_index=True, )

# Display the result
df_combined.head(-5)

surfaceProcessingLocationCavity hotRunner hotRunnerCost  \
0       55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         40000   
1       55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         40000   
2       55c4af00-0439-40aa-ab0a-4f1a0fa16882         0         40000   
3       55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         10000   
4       55c4af00-0439-40aa-ab0a-4f1a0fa16882         0         25000   
...                                      ...       ...           ...   
164140  1c1e35f0-6355-42f5-92b4-8081fcc865f2         0         40000   
164141  1c1e35f0-6355-42f5-92b4-8081fcc865f2         1         40000   
164142  55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         10000   
164143  55c4af00-0439-40aa-ab0a-4f1a0fa16882         1         10000   
164144  7c09041d-35fc-4e1a-905f-7499e3a81a8f         0         40000   

       manufacturingCost mouldValidationCost designTime hourlyDesignCost  \
0                  40000                1500         40               65   
1                  25000                3000         70               50   
2                  40000                4500        100               65   
3                  25000                4500         70               50   
4                  25000                3000         40               65   
...                  ...                 ...        ...              ...   
164140             25000                1500         40               50   
164141             55000                1500        100               50   
164142             25000                1500         40               50   
164143             25000                1500         40               50   
164144             55000                1500         70               50   

       removedChassis weightChassisProcessed  \
0                0.20                 6.6725   
1                0.15                 6.6725   
2                0.20                 6.6725   
3                0.18                 6.6725   
4                0.18                 6.6725   
...               ...                    ...   
164140           0.15                 6.6725   
164141           0.15                 6.6725   
164142           0.18                 6.6725   
164143           0.15                 6.6725   
164144           0.18                 6.6725   

            surfaceProcessingLocationChassis removedCavity numberOfCavities  \
0       35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.5               32   
1       35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.5               16   
2       35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.3               32   
3       35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.5               32   
4       35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.3               32   
...                                      ...           ...              ...   
164140  e52c3045-7a2a-44f7-9a3e-8ecc23d96345           0.3                1   
164141  e52c3045-7a2a-44f7-9a3e-8ecc23d96345           0.3                1   
164142  35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.3                1   
164143  35e16aba-caff-4fc6-a8ed-ec620c0da77f           0.3                1   
164144  d65ad7e8-09d9-47b7-a1cb-9e3c5460b9a2           0.3                1   

       weightCavityProcessed weightMould totalTransportationPercentage  \
0       0.005495000000000001    6.677995                           100   
1       0.005495000000000001    6.677995                           100   
2       0.005495000000000001    6.677995                           100   
3       0.005495000000000001    6.677995                           100   
4       0.005495000000000001    6.677995                           100   
...                      ...         ...                           ...   
164140  0.005495000000000001    6.677995                           100   
164141  0.005495000000000001    6.677995                           100   
164142  0.005495000000000001    6.677995   

In [152]:
print("Number of columns in the concatenated DataFrame:", df_combined.shape[1])
print("Number of rows in the concatenated DataFrame:", df_combined.shape[0])

Number of columns in the concatenated DataFrame: 70
Number of rows in the concatenated DataFrame: 164150


In [153]:
df= df_combined

## Saving the final combined df as csv: df_combined_F.csv

In [154]:
# Save the combined DataFrame to a CSV file using config.dest_file
output_path = os.path.join(config.dest_file, 'df_combined_F.csv')
df.to_csv(output_path, index=False)

# Prelimianry Data Analysis

## Information about df and types of data

In [155]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 164150 entries, 0 to 164149
Data columns (total 70 columns):
 #   Column                                                         Non-Null Count   Dtype  
---  ------                                                         --------------   -----  
 0   surfaceProcessingLocationCavity                                164150 non-null  object 
 1   hotRunner                                                      164150 non-null  object 
 2   hotRunnerCost                                                  164150 non-null  object 
 3   manufacturingCost                                              164150 non-null  object 
 4   mouldValidationCost                                            164150 non-null  object 
 5   designTime                                                     164150 non-null  object 
 6   hourlyDesignCost                                               164150 non-null  object 
 7   removedChassis                                 

In [156]:
df.describe()

human health - ozone layer depletion
count                          1.641500e+05
mean                           1.434746e-06
std                            1.799773e-05
min                            1.373808e-07
25%                            1.544478e-07
50%                            2.213792e-07
75%                            3.643282e-07
max                            5.529964e-03

From the above cells, it is evident that the only human health - ozone layer depletion is in float, others are objects which needs to be transformed.

## dataframe columns

In [157]:
df.columns

Index(['surfaceProcessingLocationCavity', 'hotRunner', 'hotRunnerCost',
       'manufacturingCost', 'mouldValidationCost', 'designTime',
       'hourlyDesignCost', 'removedChassis', 'weightChassisProcessed',
       'surfaceProcessingLocationChassis', 'removedCavity', 'numberOfCavities',
       'weightCavityProcessed', 'weightMould', 'totalTransportationPercentage',
       'percentageAircraft', 'percentageLorry', 'percentageTrain',
       'percentageShip', 'totalDistance', 'transportCost',
       'injectedMaterial_product', 'injectedMaterialCost',
       'percentageRecycledMaterial', 'maxDepth', 'maxWallThickness',
       'productVolume', 'materozzaVolume', 'nAnniProduzione', 'nProdottiAnno',
       'materialDensity', 'tolerance', 'surfaceFinishing', 'cycleTime',
       'machineCycleTime', 'maintenanceCost', 'productionCost',
       'injectedMaterial_materozza', 'injectionMouldingProcess',
       'memtiEngineValue', 'steelPrice', 'runnersType', 'mouldMaterialName',
       'machineName',

## Dealing with non-numerical columns using Encoding

I have found that some columns have non-numrical values. It is important to identify these coulmns to treat them accordingly for further analysis.


In [158]:
nonnumerical_columns = df.select_dtypes(exclude=[np.number]).columns.tolist()
print("Non-numerical columns in df:", nonnumerical_columns)

Non-numerical columns in df: ['surfaceProcessingLocationCavity', 'hotRunner', 'hotRunnerCost', 'manufacturingCost', 'mouldValidationCost', 'designTime', 'hourlyDesignCost', 'removedChassis', 'weightChassisProcessed', 'surfaceProcessingLocationChassis', 'removedCavity', 'numberOfCavities', 'weightCavityProcessed', 'weightMould', 'totalTransportationPercentage', 'percentageAircraft', 'percentageLorry', 'percentageTrain', 'percentageShip', 'totalDistance', 'transportCost', 'injectedMaterial_product', 'injectedMaterialCost', 'percentageRecycledMaterial', 'maxDepth', 'maxWallThickness', 'productVolume', 'materozzaVolume', 'nAnniProduzione', 'nProdottiAnno', 'materialDensity', 'tolerance', 'surfaceFinishing', 'cycleTime', 'machineCycleTime', 'maintenanceCost', 'productionCost', 'injectedMaterial_materozza', 'injectionMouldingProcess', 'memtiEngineValue', 'steelPrice', 'runnersType', 'mouldMaterialName', 'machineName', 'EUUSMacchina', 'CNMacchina', 'gateDiameter', 'setupTime', 'warmupTime', '

Since in non-numerical columns, there are features and target columns, we will separate them using the exert knowledge that columns from cost onwards are targets, which needed to be convert.

In [159]:

features_to_convert = nonnumerical_columns[:-16] # Exclude the last 16 columns which are likely target columns
targets_to_convert = nonnumerical_columns[-16:]  # Last 16 columns are likely target columns
print("Features to convert:", features_to_convert)
print("Targets to convert:", targets_to_convert)


Features to convert: ['surfaceProcessingLocationCavity', 'hotRunner', 'hotRunnerCost', 'manufacturingCost', 'mouldValidationCost', 'designTime', 'hourlyDesignCost', 'removedChassis', 'weightChassisProcessed', 'surfaceProcessingLocationChassis', 'removedCavity', 'numberOfCavities', 'weightCavityProcessed', 'weightMould', 'totalTransportationPercentage', 'percentageAircraft', 'percentageLorry', 'percentageTrain', 'percentageShip', 'totalDistance', 'transportCost', 'injectedMaterial_product', 'injectedMaterialCost', 'percentageRecycledMaterial', 'maxDepth', 'maxWallThickness', 'productVolume', 'materozzaVolume', 'nAnniProduzione', 'nProdottiAnno', 'materialDensity', 'tolerance', 'surfaceFinishing', 'cycleTime', 'machineCycleTime', 'maintenanceCost', 'productionCost', 'injectedMaterial_materozza', 'injectionMouldingProcess', 'memtiEngineValue', 'steelPrice', 'runnersType', 'mouldMaterialName', 'machineName', 'EUUSMacchina', 'CNMacchina', 'gateDiameter', 'setupTime', 'warmupTime', 'delivery

In [160]:
df[df.columns] = df[df.columns].astype(str)

In [161]:
# Identify categorical and continuous non-numerical columns, then plot only categorical ones

categorical_cols = []
continuous_cols = []

for col in nonnumerical_columns:
    unique_count = df[col].nunique()
    if unique_count < 20:
        categorical_cols.append(col)
    else:
        continuous_cols.append(col)

print("Categorical columns:", categorical_cols)
print("Continuous or high-cardinality columns:", continuous_cols)



Categorical columns: ['surfaceProcessingLocationCavity', 'hotRunner', 'hotRunnerCost', 'manufacturingCost', 'mouldValidationCost', 'designTime', 'hourlyDesignCost', 'removedChassis', 'weightChassisProcessed', 'surfaceProcessingLocationChassis', 'removedCavity', 'numberOfCavities', 'weightCavityProcessed', 'weightMould', 'totalTransportationPercentage', 'percentageAircraft', 'percentageLorry', 'percentageTrain', 'percentageShip', 'totalDistance', 'transportCost', 'injectedMaterial_product', 'injectedMaterialCost', 'percentageRecycledMaterial', 'maxDepth', 'maxWallThickness', 'productVolume', 'materozzaVolume', 'nAnniProduzione', 'nProdottiAnno', 'materialDensity', 'tolerance', 'surfaceFinishing', 'cycleTime', 'machineCycleTime', 'maintenanceCost', 'productionCost', 'injectedMaterial_materozza', 'injectionMouldingProcess', 'memtiEngineValue', 'steelPrice', 'runnersType', 'mouldMaterialName', 'EUUSMacchina', 'CNMacchina', 'gateDiameter', 'setupTime', 'warmupTime', 'deliveryVolume', 'deliv

In [162]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns
import numpy as np
from sklearn.linear_model import LinearRegression, Ridge

In [163]:
columns_to_convert = ['surfaceProcessingLocationCavity', 'surfaceProcessingLocationChassis', 
                      'injectedMaterial_product', 'injectedMaterial_materozza', 
                      'injectionMouldingProcess','memtiEngineValue', 'steelPrice', 
                      'runnersType', 'mouldMaterialName','machineName', 
                      'EUUSMacchina', 'CNMacchina']

target_column = ['Cost',
       'human health - photochemical oxidation',
       'ecosystem quality - terrestrial ecotoxicity',
       'resources - mineral extraction', 'resources - non-renewable energy',
       'ecosystem quality - terrestrial acidification & nutrification',
       'resources - total', 'human health - ionising radiation',
       'human health - respiratory effects (inorganics)',
       'human health - total', 'human health - human toxicity',
       'ecosystem quality - aquatic ecotoxicity',
       'climate change - climate change',
       'human health - ozone layer depletion',
       'ecosystem quality - land occupation', 'climate change - total',
       'ecosystem quality - total']



#df[columns_to_convert] = df[columns_to_convert].astype(str)
df[df.columns] = df[df.columns].astype(str)

In [164]:
label_encoder = LabelEncoder()
df_encoded= df.copy()

In [165]:
for col in features_to_convert:
    df_encoded[col] = label_encoder.fit_transform(df_encoded[col]).astype(float)

In [166]:
feature_column = ['surfaceProcessingLocationCavity', 'hotRunner', 'hotRunnerCost',
       'manufacturingCost', 'mouldValidationCost', 'designTime',
       'hourlyDesignCost', 'removedChassis', 'weightChassisProcessed',
       'surfaceProcessingLocationChassis', 'removedCavity', 'numberOfCavities',
       'weightCavityProcessed', 'weightMould', 'totalTransportationPercentage',
       'percentageAircraft', 'percentageLorry', 'percentageTrain',
       'percentageShip', 'totalDistance', 'transportCost',
       'injectedMaterial_product', 'injectedMaterialCost',
       'percentageRecycledMaterial', 'maxDepth', 'maxWallThickness',
       'productVolume', 'materozzaVolume', 'nAnniProduzione', 'nProdottiAnno',
       'materialDensity', 'tolerance', 'surfaceFinishing', 'cycleTime',
       'machineCycleTime', 'maintenanceCost', 'productionCost',
       'injectedMaterial_materozza', 'injectionMouldingProcess',
       'memtiEngineValue', 'steelPrice', 'runnersType', 'mouldMaterialName',
       'machineName', 'EUUSMacchina', 'CNMacchina', 'gateDiameter',
       'setupTime', 'warmupTime', 'deliveryVolume', 'deliveryPeriod',
       'mouldDesignCostDisplay', 'mouldTotalCost']

target_column = ['Cost',
       'human health - photochemical oxidation',
       'ecosystem quality - terrestrial ecotoxicity',
       'resources - mineral extraction', 'resources - non-renewable energy',
       'ecosystem quality - terrestrial acidification & nutrification',
       'resources - total', 'human health - ionising radiation',
       'human health - respiratory effects (inorganics)',
       'human health - total', 'human health - human toxicity',
       'ecosystem quality - aquatic ecotoxicity',
       'climate change - climate change',
       'human health - ozone layer depletion',
       'ecosystem quality - land occupation', 'climate change - total',
       'ecosystem quality - total']

In [167]:
df_encoded.head(-5)

surfaceProcessingLocationCavity  hotRunner  hotRunnerCost  \
0                                   3.0        1.0            3.0   
1                                   3.0        1.0            3.0   
2                                   3.0        0.0            3.0   
3                                   3.0        1.0            1.0   
4                                   3.0        0.0            2.0   
...                                 ...        ...            ...   
164140                              2.0        0.0            3.0   
164141                              2.0        1.0            3.0   
164142                              3.0        1.0            1.0   
164143                              3.0        1.0            1.0   
164144                              4.0        0.0            3.0   

        manufacturingCost  mouldValidationCost  designTime  hourlyDesignCost  \
0                     2.0                  0.0         3.0               2.0   
1                     1.0                  2.0         4.0               1.0   
2                     2.0                  3.0         2.0               2.0   
3                     1.0                  3.0         4.0               1.0   
4                     1.0                  2.0         3.0               2.0   
...                   ...                  ...         ...               ...   
164140                1.0                  0.0         3.0               1.0   
164141                3.0                  0.0         2.0               1.0   
164142                1.0                  0.0         3.0               1.0   
164143                1.0                  0.0         3.0               1.0   
164144                3.0                  0.0         4.0               1.0   

        removedChassis  weightChassisProcessed  \
0                  3.0                     1.0   
1                  1.0                     1.0   
2                  3.0                     1.0   
3                  2.0                     1.0   
4                  2.0                     1.0   
...                ...                     ...   
164140             1.0                     1.0   
164141             1.0                     1.0   
164142             2.0                     1.0   
164143             1.0                     1.0   
164144             2.0                     1.0   

        surfaceProcessingLocationChassis  removedCavity  numberOfCavities  \
0                                    0.0            4.0               3.0   
1                                    0.0            4.0               1.0   
2                                    0.0            2.0               3.0   
3                                    0.0            4.0               3.0   
4                                    0.0            2.0               3.0   
...                                  ...            ...               ...   
164140                               3.0            2.0               0.0   
164141                               3.0            2.0               0.0   
164142                               0.0            2.0               0.0   
164143                               0.0            2.0               0.0   
164144                               2.0            2.0               0.0   

        weightCavityProcessed  weightMould  totalTransportationPercentage  \
0                         2.0          2.0                            0.0   
1                         2.0          2.0                            0.0   
2                         2.0          2.0                            0.0   
3                         2.0          2.0                            0.0   
4                         2.0          2.0                            0.0   
...                       ...          ...                            ...   
164140                    2.0          2.0                            0.0   
164141                    2.0          2.0                            0.0   

In [168]:
# Check if the value exists anywhere in the DataFrame
value = 'Horizontal 10 ton'
exists = (df_encoded == value).any().any()

print(f"Value exists: {exists}")
positions = df_encoded.stack()[df_encoded.stack() == value].index

# Print positions
for pos in positions:
    print(f"Value found at: {pos}")

Value exists: True
Value found at: (53946, 'Cost')
Value found at: (53947, 'Cost')


In [169]:
# Get distinct values in the 'Snowfall' column
distinct_values = df_encoded['percentageLorry'].unique()
distinct_values

array([0., 1.])

In [170]:
df_encoded.drop(df_encoded[df_encoded['percentageLorry'] == 'Horizontal 10 ton'].index, inplace=True)

In [172]:
df_encoded.drop(index=53946, inplace=True)  # Corrected index to drop
df_encoded.drop(index=53947, inplace=True)  # Corrected index to drop

In [173]:
df_encoded[df_encoded.columns] = df_encoded[df_encoded.columns].astype(float)

In [175]:
df_encoded.head(53950)

surfaceProcessingLocationCavity  hotRunner  hotRunnerCost  \
0                                  3.0        1.0            3.0   
1                                  3.0        1.0            3.0   
2                                  3.0        0.0            3.0   
3                                  3.0        1.0            1.0   
4                                  3.0        0.0            2.0   
...                                ...        ...            ...   
53945                              3.0        0.0            1.0   
53948                              3.0        0.0            1.0   
53949                              3.0        0.0            1.0   
53950                              3.0        0.0            1.0   
53951                              3.0        0.0            1.0   

       manufacturingCost  mouldValidationCost  designTime  hourlyDesignCost  \
0                    2.0                  0.0         3.0               2.0   
1                    1.0                  2.0         4.0               1.0   
2                    2.0                  3.0         2.0               2.0   
3                    1.0                  3.0         4.0               1.0   
4                    1.0                  2.0         3.0               2.0   
...                  ...                  ...         ...               ...   
53945                3.0                  0.0         3.0               1.0   
53948                3.0                  0.0         3.0               1.0   
53949                3.0                  0.0         3.0               1.0   
53950                3.0                  0.0         3.0               1.0   
53951                3.0                  0.0         3.0               1.0   

       removedChassis  weightChassisProcessed  \
0                 3.0                     1.0   
1                 1.0                     1.0   
2                 3.0                     1.0   
3                 2.0                     1.0   
4                 2.0                     1.0   
...               ...                     ...   
53945             1.0                     1.0   
53948             1.0                     1.0   
53949             1.0                     1.0   
53950             1.0                     1.0   
53951             1.0                     1.0   

       surfaceProcessingLocationChassis  removedCavity  numberOfCavities  \
0                                   0.0            4.0               3.0   
1                                   0.0            4.0               1.0   
2                                   0.0            2.0               3.0   
3                                   0.0            4.0               3.0   
4                                   0.0            2.0               3.0   
...                                 ...            ...               ...   
53945                               0.0            2.0               3.0   
53948                               0.0            2.0               0.0   
53949                               0.0            2.0               0.0   
53950                               0.0            2.0               0.0   
53951                               0.0            2.0               0.0   

       weightCavityProcessed  weightMould  totalTransportationPercentage  \
0                        2.0          2.0                            0.0   
1                        2.0          2.0                            0.0   
2                        2.0          2.0                            0.0   
3                        2.0          2.0                            0.0   
4                        2.0          2.0                            0.0   
...                      ...          ...                            ...   
53945                    2.0          2.0                            0.0   
53948                    2.0          2.0                            0.0   
53949                    2.0          2.0              

In [177]:
dest_path = os.path.join(config.dest_file, "df_encoded.csv")
df_encoded.to_csv(dest_path, index=False)